### PR TITLE
add productInfoAttr

### PR DESCRIPTION
**Pull request by realwanpengli**

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Common/Utilities/ProductInfoAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.Azure.SignalR.Common
+{
+    [AttributeUsage(AttributeTargets.Assembly)]
+    class ProductInfoAttribute : Attribute
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR.Common/Utilities/ProductInfoAttribute.cs]: add `internal` explicitly

File: src/Microsoft.Azure.SignalR/ProductInfo.cs [hunk 2]
```diff
@@ -1,22 +1,49 @@
...
 
 namespace Microsoft.Azure.SignalR
 {
     internal static class ProductInfo
     {
+        private const int MaxProductInfoLength = 256;
         public static string GetProductInfo()
         {
-            var assembly = Assembly.GetCallingAssembly();
+            var allAssemblies = AppDomain.CurrentDomain.GetAssemblies();
```
Comment (vicancy) [inline src/Microsoft.Azure.SignalR/ProductInfo.cs]: Maybe just make it for Management API?

File: src/Microsoft.Azure.SignalR/ProductInfo.cs [hunk 3]
```diff
@@ -1,22 +1,49 @@
...
+
+        public static string GetProductInfoCore(Assembly assembly)
+        {
             var packageId = assembly.GetName().Name;
             var version = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
             var runtime = RuntimeInformation.FrameworkDescription?.Trim();
             var operatingSystem = RuntimeInformation.OSDescription?.Trim();
             var processorArchitecture = RuntimeInformation.ProcessArchitecture.ToString().Trim();
-            
             return $"{packageId}/{version} ({runtime}; {operatingSystem}; {processorArchitecture})";
```
Comment (JialinXin) [inline src/Microsoft.Azure.SignalR/ProductInfo.cs]: >({runtime}; {operatingSystem}; {processorArchitecture}) [](start = 43, length = 55)

For multiple assembilies, the runtime information will be duplicated?

File: src/Microsoft.Azure.SignalR/ProductInfo.cs (hunk 2 above)
Comment (realwanpengli) [reply src/Microsoft.Azure.SignalR/ProductInfo.cs]: But management SDK reuse service SDK, so there seems no way to tell which one is quering product info if we don't change api.

**Comment by realwanpengli** [comment]
cons: what if someone use both service sdk and management sdk?

discussed and figure out a better solution: add a method that input a assembly in builder,

**Pull request closed without merging**